src/bot/core/music.py

```
@@ -0,0 +1,466 @@
+        self.length = content[number]["videoRenderer"]["lengthText"]["simpleText"]
+
+
+class Music(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+        self._last_member = None
+
+    def setPlayer(self, ctx):
+        guild_id = ctx.message.guild.id
+        try:
+            songQueue = song_list_dict[guild_id][0]
+            song_list = song_list_dict[guild_id][1]
+        except KeyError:
+            song_list_dict[guild_id] = [0, list()]
+            songQueue = song_list_dict[guild_id][0]
+            song_list = song_list_dict[guild_id][1]
+
+        return songQueue, song_list
+
+    def updatePlayer(self, ctx, song_list, songQueue):
+        guild_id = ctx.message.guild.id
+        song_list_dict[guild_id][0] = songQueue
+        song_list_dict[guild_id][1] = song_list
+
+    async def player(self, ctx: commands.Context):
+        if ctx.voice_client == None:
+            if ctx.author.voice:
+                await ctx.author.voice.channel.connect()
+                await self.playMusic(ctx)
+            else:
+                await ctx.send(
+                    embed=MsgFormatter.get(
+                        ctx,
+                        "Usage Error",
+                        "You are not in any voice channel. Please join a voice channel to use Music bot.",
+                    )
+                )
+                raise commands.CommandError("Author not connected to a voice channel.")
+
+    async def playMusic(self, ctx):
+        songQueue, song_list = self.setPlayer(ctx)
+
+        def next():
+            songQueue, song_list = self.setPlayer(ctx)
+            if len(song_list) > songQueue:
+                songQueue += 1
+                fut = asyncio.run_coroutine_threadsafe(
+                    self.playMusic(ctx), self.bot.loop
+                )
+                try:
+                    fut.result()
+                    self.updatePlayer(song_list, songQueue)
+                except:
+                    pass
```

```suggestion
                fut.result()
                self.updatePlayer(ctx, song_list, songQueue)
```
We should not do these `try: ... except:`, because it hides errors and would make things difficult to debug.

---

Excepting try - except cycle raises error